addons/project_timesheet_holidays/models/project_task.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+class Task(models.Model):
+    _inherit = 'project.task'
+
+    timeoff_type_ids = fields.One2many('hr.leave.type', 'timesheet_task_id', string="Task Time off Types")
```

Comment:
I think we could rename the field into `leave_type_ids` since the model is `hr.leave.type`. 🤔